accumulator/accumulate_signals_workflow.go

```diff
@@ -0,0 +1,168 @@
+package accumulator
+
+import (
+	"strconv"
+	"time"
+
+	"go.temporal.io/sdk/workflow"
+)
+
+/**
+ * This sample demonstrates how to accumulate many signals (business events) over a time period.
+ * This sample implements the Accumulator Pattern: collect many meaningful things that
+ *   need to be collected and worked on together, such as all payments for an account, or
+ *   all account updates by account.
+ * This sample models robots being created throughout the time period,
+ *   groups them by what color they are, and greets all the robots of a color at the end.
+ *
+ * A new workflow is created per grouping. Workflows continue as new as needed.
+ * A sample activity at the end is given, and you could add an activity to
+ *   process individual events in the processGreeting() method.
+ *
+ * Because Temporal Workflows cannot have an unlimited size, Continue As New is used
+ *   to process more signals that may come in.
+ * You could create as many groupings as desired, as Temporal Workflows scale out to many workflows without limit.
+ * You could vary the time that the workflow waits for other signals, say for a day, or a variable time from first
+ *   signal with the GetNextTimeout() function.
+ */
+
+// signalToSignalTimeout is them maximum time between signals
+const signalToSignalTimeout = 30 * time.Second
+
+// fromStartTimeout is the maximum time to receive all signals
+const fromStartTimeout = 60 * time.Second
+
+// exitTimeout is the time to wait after exit is requested to catch any last few signals
+const exitTimeout = 1 * time.Second
+
+type AccumulateGreeting struct {
+	GreetingText string
+	Bucket       string
+	GreetingKey  string
+}
+
+type GreetingsInfo struct {
+	BucketKey          string
+	GreetingsList      []AccumulateGreeting
+	UniqueGreetingKeys map[string]bool
+	startTime          time.Time
+}
+
+// GetNextTimeout returns the maximum time for a workflow to wait for the next signal.
+// This waits for the greater of the remaining fromStartTimeout and  signalToSignalTimeout
+// fromStartTimeout and signalToSignalTimeout can be adjusted to wait for the right amount of time as desired
+// This resets with Continue As New
+func (a *AccumulateGreeting) GetNextTimeout(ctx workflow.Context, startTime time.Time, exitRequested bool) (time.Duration, error) {
+	if exitRequested {
+		return exitTimeout, nil
+	}
+	if startTime.IsZero() {
+		startTime = workflow.GetInfo(ctx).WorkflowStartTime // if you want to start from the time of the first signal, customize this
+	}
+	total := workflow.Now(ctx).Sub(startTime)
+	totalLeft := fromStartTimeout - total
+	if totalLeft <= 0 {
+		return 0, nil
+	}
+	if signalToSignalTimeout > totalLeft {
```

Comment:
Don't you mean:

```suggestion
	if totalLeft > signalToSignalTimeout {
```

Since `signalToSignalTimeout` says it's the "maximum"?